code/parachain/frame/crowdloan-rewards/src/lib.rs

```diff
@@ -350,6 +396,49 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
+		/// Patch existing rewards by deleting them, and adds the the `replacements` section.
+		/// This will cause inconsistencies if a claim has already been made. Errors can be avoided
+		/// by passing the expected amount of rewards claimable.
+		pub(crate) fn do_delete(
+			deletions: Vec<(RemoteAccountOf<T>, BalanceOf<T>)>,
+		) -> DispatchResult {
+			let total_rewards: T::Balance = TotalRewards::<T>::get();
+			let total_contributors: u32 = TotalContributors::<T>::get();
+
+			// Get each account from deletions and remove it from the storage
+			for (account, expected) in deletions {
+				Rewards::<T>::mutate_exists(account, |rewards| {
+					if let Some(rewards) = rewards {
+						ensure!(rewards.total == expected, Error::<T>::UnexpectedRewardAmount);
+						TotalRewards::<T>::mutate(|total| *total -= rewards.total);
+						TotalContributors::<T>::mutate(|total| *total -= 1);
+					}
+					*rewards = None;
+					Ok::<(), Error<T>>(())
+				})?;
+			}
+
+			TotalRewards::<T>::set(total_rewards);
+			TotalContributors::<T>::set(total_contributors);
+			Ok(())
+		}
+
+		/// Add new rewards to the pallet. Updates total_rewards and contributors accordingly.
+		pub fn do_add(
+			additions: Vec<(RemoteAccountOf<T>, RewardAmountOf<T>, VestingPeriodOf<T>)>,
```

Comment:
I think we can avoid adding a delete by using the new reward amounts as 0 for removed ones. Saves much of the changes needed.

Reply:
So two separate extrinsics, one doing do_delete // set rewards to 0.
Or confounding them in 1 extrinsic? I initially had the do_replacements extrinsic perform both operations, but that was a bit messy. `delete` and `add` will be called within utility.batch anyway.